packages/fetcher-atama/index.ts

```diff
@@ -95,28 +105,74 @@ export class FetcherAtama extends Fetcher<AtamaFetcherConfig> {
    * @param {string} identifier The path to get data for
    */
   async getData<T>(identifier: string): Promise<CXExperience<T>> {
-    const url = new URL(
-      `v1/${this.config.environment || 'prod'}/${
-        this.config.workspaceId
-      }/data/${identifier.startsWith('/') ? identifier.slice(1) : identifier}`,
-      this.config.url || 'https://cdn.atama.land',
-    );
+    const dataInput: GetDataInput = {
+      workspaceId: this.config.workspaceId,
+      environment: this.config.environment || 'prod',
+      slug: identifier,
+    };
 
     try {
-      // eslint-disable-next-line no-console
-      console.debug(`Running request to get data against ${url.toString()}`);
-
-      const response = await fetch(url.toString(), {
-        headers: {
-          Authorization: `Bearer ${this.config.apiKey}`,
+      const response = await this.runGraphQLRequest<{
+        getData: CXExperience<T>;
+      }>(
+        gql`
+          query GetData($dataInput: GetDataInput!) {
+            getData(dataInput: $dataInput) {
+              meta
+              template
+              placements {
+                code
+                embeddableBlueprint {
+                  template
+                  placements {
+                    components {
+                      correlationId
+                      type
+                      contentProperties
+                      visualProperties
+                      componentTypeName
+                    }
+                  }
+                }
+                components {
+                  correlationId
+                  type
+                  contentProperties
+                  visualProperties
+                  componentTypeName
+                }
+              }
+              bundleManifest {
+                providers {
+                  type
+                  endpoint
+                }
+                actions {
+                  actionId
+                  providerConfigs {
+                    providerConfigSecretArn
+                    type
+                    mappingDefinitions {
+                      key
+                      type
+                      instructions
+                    }
+                  }
+                }
+              }
+            }
+          }
+        `,
+        {
+          dataInput,
         },
-      });
+      );
 
-      if (!response.ok) {
-        throw new AtamaFetcherError(response.status);
+      if (!response.getData) {
+        throw new AtamaFetcherError(500);
       }
```

Comment:
In the case of getting data for a specific path we need to be more nuanced with the status code. We need to differentiate between "the API doesn't respond / throwing an error" => 500 and "the API responded but it couldn't find data for the given path" => 404.
This is important so the framework consuming this information can continue serving data from the cache (in the case of a 500, and if it actually has something cached for the given path), show a 500 error page (in case of a 500, if no data was cached before), or show a 404 error page.